awx/main/tasks.py

```diff
@@ -407,7 +405,26 @@ def cleanup_execution_environment_images():
 
 @task(queue=get_local_queuename)
 def check_heartbeat(node):
-    AWXReceptorJob.check_heartbeat(node)
+    instance = Instance.objects.get(hostname=node)
+    data = worker_info(node)
+
+    if data['Errors']:
+        formatted_error = "\n".join(data["Errors"])
+        logger.warn(f'Failed to find capacity of execution node {node}, errors:\n{formatted_error}')
+    else:
+        # TODO: spin off new instance method from refresh_capacity that calculates derived fields
```

Comment:
@jladdjr I was trying to talk to you about a `refresh_capacity` refactor. The concept is there is a system check which produces in fields `cpu` and `memory`, which are in some kind of natural units, like "number of CPUs (varies by counting method)" or "bytes".

Then we need another method which is clearly logically separated to go from natural units to `cpu_capacity`, `mem_capacity`, and obviously `capacity`. These are in a form of arbitrary units. They're something the task manager uses to count stuff so it figures out where to put jobs. I want these to always appear in code in two separate processes, even if control plane nodes do them both in one go.